### PR TITLE
Report progress from server-side to keep client connections alive

### PR DESCRIPTION
**Pull request by abyss7**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Improvement

This way we make sure that 'receive_timeout' on client-side is triggered only when there is a real socket read timeout.

### CODE REVIEW

**Comment by abyss7** [comment]
The `stateless.00411_long_accurate_number_comparison` has timed out on Jenkins, while passed fine on Sandbox.

File: dbms/programs/server/TCPHandler.cpp [hunk 1]
```diff
@@ -409,7 +409,7 @@ void TCPHandler::processOrdinaryQuery()
                 }
                 else
                 {
-                    if (state.progress.rows && after_send_progress.elapsed() / 1000 >= query_context.getSettingsRef().interactive_delay)
+                    if (after_send_progress.elapsed() / 1000 >= query_context.getSettingsRef().interactive_delay)
```
[the remark this answers is not in the document]
Comment (abyss7) [reply dbms/programs/server/TCPHandler.cpp]: Fixed

File: dbms/programs/client/Client.cpp [hunk 2]
```diff
@@ -1648,9 +1683,12 @@ class Client : public Poco::Util::Application
...
 
         /// Extract settings from the options.
-#define EXTRACT_SETTING(TYPE, NAME, DEFAULT, DESCRIPTION) \
-        if (options.count(#NAME)) \
-            context.setSetting(#NAME, options[#NAME].as<std::string>());
+#define EXTRACT_SETTING(TYPE, NAME, DEFAULT, DESCRIPTION)                \
+        if (options.count(#NAME))                                        \
+        {                                                                \
+            context.setSetting(#NAME, options[#NAME].as<std::string>()); \
+            config().setString(#NAME, options[#NAME].as<std::string>()); \
```
Comment (vitlibar) [inline dbms/programs/client/Client.cpp]: @abyss7 why do we put settings to `config()` here ? Looks excessive.
Maybe you actually wanted a single setting to be in `config()`, not everything?

File: dbms/programs/server/TCPHandler.cpp [hunk 3]
```diff
@@ -409,7 +409,7 @@ void TCPHandler::processOrdinaryQuery()
...
                     {
                         /// Some time passed and there is a progress.
```
Comment (alexey-milovidov) [inline dbms/programs/server/TCPHandler.cpp]: Obsolete comment.

**Pull request merged**